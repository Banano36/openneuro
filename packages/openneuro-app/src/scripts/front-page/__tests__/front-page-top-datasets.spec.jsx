import React from 'react'
import { shallow } from 'enzyme'
import FrontPageTopDatasets, {
  FrontPageTopResult,
} from '../front-page-top-datasets.jsx'

describe('FrontPageTopDatasets', () => {
  it('renders container correctly', () => {
    const wrapper = shallow(<FrontPageTopDatasets />)
    expect(wrapper).toMatchSnapshot()
  })
  describe('FrontPageTopResult', () => {
    it('does not crash with null query results', () => {
      const fakeQuery = Symbol('not a real query')
      const renderFrontPageTopResult = () => {
        shallow(
          FrontPageTopResult(fakeQuery)({
            loading: false,
            error: false,
            data: {},
          }),
        )
      }
      expect(renderFrontPageTopResult).not.toThrowError()
    })
  })
})
