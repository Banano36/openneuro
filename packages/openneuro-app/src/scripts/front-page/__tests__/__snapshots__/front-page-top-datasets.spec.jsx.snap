// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPageTopDatasets renders container correctly 1`] = `
<Fragment>
  <div
    className="browse-pipelines"
  >
    <h3
      className="browse-pipeline-header"
    >
      Recent Activity
    </h3>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <Styled(div)>
          <div
            className="col-sm-6 mate-slide fade-in"
          >
            <h4>
              Most Active
            </h4>
            <FrontPageTopQuery
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "top_viewed_datasets",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "first",
                                },
                                "value": Object {
                                  "kind": "IntValue",
                                  "value": "5",
                                },
                              },
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orderBy",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "views",
                                      },
                                      "value": Object {
                                        "kind": "EnumValue",
                                        "value": "descending",
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "filterBy",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "public",
                                      },
                                      "value": Object {
                                        "kind": "BooleanValue",
                                        "value": true,
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "datasets",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "edges",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "node",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "analytics",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "alias": undefined,
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "views",
                                                    },
                                                    "selectionSet": undefined,
                                                  },
                                                ],
                                              },
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "latestSnapshot",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "alias": undefined,
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "tag",
                                                    },
                                                    "selectionSet": undefined,
                                                  },
                                                  Object {
                                                    "alias": undefined,
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "description",
                                                    },
                                                    "selectionSet": Object {
                                                      "kind": "SelectionSet",
                                                      "selections": Array [
                                                        Object {
                                                          "alias": undefined,
                                                          "arguments": Array [],
                                                          "directives": Array [],
                                                          "kind": "Field",
                                                          "name": Object {
                                                            "kind": "Name",
                                                            "value": "Name",
                                                          },
                                                          "selectionSet": undefined,
                                                        },
                                                      ],
                                                    },
                                                  },
                                                ],
                                              },
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 374,
                    "start": 0,
                  },
                }
              }
            />
          </div>
          <div
            className="col-sm-6 mate-slide fade-in browse"
          >
            <h4>
              Recently Published
            </h4>
            <FrontPageTopQuery
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "recently_published_datasets",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "first",
                                },
                                "value": Object {
                                  "kind": "IntValue",
                                  "value": "5",
                                },
                              },
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orderBy",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "publishDate",
                                      },
                                      "value": Object {
                                        "kind": "EnumValue",
                                        "value": "descending",
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "filterBy",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "public",
                                      },
                                      "value": Object {
                                        "kind": "BooleanValue",
                                        "value": true,
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "datasets",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "edges",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "node",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "publishDate",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "latestSnapshot",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "alias": undefined,
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "tag",
                                                    },
                                                    "selectionSet": undefined,
                                                  },
                                                  Object {
                                                    "alias": undefined,
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "description",
                                                    },
                                                    "selectionSet": Object {
                                                      "kind": "SelectionSet",
                                                      "selections": Array [
                                                        Object {
                                                          "alias": undefined,
                                                          "arguments": Array [],
                                                          "directives": Array [],
                                                          "kind": "Field",
                                                          "name": Object {
                                                            "kind": "Name",
                                                            "value": "Name",
                                                          },
                                                          "selectionSet": undefined,
                                                        },
                                                      ],
                                                    },
                                                  },
                                                ],
                                              },
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 358,
                    "start": 0,
                  },
                }
              }
            />
          </div>
        </Styled(div)>
      </div>
    </div>
  </div>
</Fragment>
`;
